apps/verticalhome/js/app.js

```diff
@@ -150,6 +178,191 @@
 
     },
 
+    saveAllPinnedApps: function(){
+      var pinnedItems= app.pinnedAppsManager.items;
+      if(this.deletedApps.length>0){
+        for(var x = 0; x < pinnedItems.length; x++){
+          if(pinnedItems[x].element.style.display == 'none'){
+            pinnedItems.splice(x,1);
+          }
+        }
+
+        for(var z = 0; z < this.deletedApps.length; z++){
+          app.itemStore.deletePinnedAppItem(this.deletedApps[z]);
+        }
+      }
+
+        for(var i = 0; i<pinnedItems.length; i++){
+          var obj = {};
+
+          if(pinnedItems[i].entry!=null){
+
+              obj.name = pinnedItems[i].entry.name;
+              obj.manifestURL = typeof(
+                pinnedItems[i].entry.manifestURL
+                ) == 'undefined'?
+                pinnedItems[i].entry.targetApp.manifestURL
+                : pinnedItems[i].entry.manifestURL;
+
+              obj.icon = typeof(
+                pinnedItems[i].entry.icon
+                )=='undefined'?
+                pinnedItems[i].entry.icon.baseURI
+                :pinnedItems[i].entry.icon;
+
+              obj.index = pinnedItems[i].index;
+
+              app.itemStore.savePinnedAppItem(obj);
+
+          }
+        }
+
+       this.showNotif('Changes saved');
+
+    },
+
+    pinToMainScreen: function(){
+      var moreAppsSelected = '#more-apps-screen .selected';
+      var selectedMoreAppItem = document.querySelector(moreAppsSelected);
+
+      var selectedMoreAppItem_icon = selectedMoreAppItem
+        .getAttribute('data-test-icon');
+
+      var selectedMoreAppItem_data_identifier = selectedMoreAppItem
+        .getAttribute('data-identifier');
+
+      var selectedMoreAppItem_title = selectedMoreAppItem
+        .querySelector('span.title').textContent;
+
+      var notif = document.getElementById('blinknotif');
```

Comment:
It seems this variable isn't used. The same for this variable in unpinFromMainScreen() function